plugins/modules/ec2_asg_tag.py

```diff
@@ -0,0 +1,237 @@
+      - environment: development
+
+- name: Remove all tags except Name from an ASG
+  community.aws.ec2_asg_tag:
+    name: my-auto-scaling-group
+    state: absent
+    tags:
+      - Name: ''
+    purge_tags: true
+'''
+
+RETURN = r'''
+---
+tags:
+  description: A list containing the tags on the resource
+  returned: always
+  type: list
+  sample: [
+      {
+          "key": "Name",
+          "value": "public-webapp-production-1",
+          "resource_id": "public-webapp-production-1",
+          "resource_type": "auto-scaling-group",
+          "propagate_at_launch": "true"
+      },
+      {
+          "key": "env",
+          "value": "production",
+          "resource_id": "public-webapp-production-1",
+          "resource_type": "auto-scaling-group",
+          "propagate_at_launch": "true"
+      }
+  ]
+added_tags:
+  description: A list of tags that were added to the ASG
+  returned: If tags were added
+  type: list
+removed_tags:
+  description: A list of tags that were removed from the ASG
+  returned: If tags were removed
+  type: list
+'''
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # Handled by AnsibleAWSModule
+
+from ansible.module_utils._text import to_native
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+
+
+def to_boto3_tag_list(tags, group_name):
+    tag_list = []
+    for tag in tags:
+        for k, v in tag.items():
+            if k == 'propagate_at_launch':
+                continue
+            tag_list.append(dict(Key=k,
+                                 Value=to_native(v) if v else v,
+                                 PropagateAtLaunch=bool(tag.get('propagate_at_launch', True)),
+                                 ResourceType='auto-scaling-group',
+                                 ResourceId=group_name))
+    return tag_list
+
+
+def compare_asg_tags(current_tags_dict, new_tags_dict, purge_tags=True):
+    """
+    Compare two ASG tag dicts.
+
+    :param current_tags_dict: dict of currently defined boto3 tags.
+    :param new_tags_dict: dict of new boto3 tags to apply.
+    :param purge_tags: whether to consider tags not in new_tags_dict for removal.
+    :return: tags_to_set: a dict of boto3 tags to set. If all tags are identical this list will be empty.
+    :return: tags_keys_to_unset: a list of tag keys to be unset. If no tags need to be unset this list will be empty.
+    """
+
+    tags_to_set = {}
+    tag_keys_to_unset = []
+
+    for key in current_tags_dict.keys():
+        if key not in new_tags_dict and purge_tags:
+            tag_keys_to_unset.append(key)
+
+    for key in set(new_tags_dict.keys()) - set(tag_keys_to_unset):
+        if new_tags_dict[key] != current_tags_dict.get(key):
+            tags_to_set[key] = new_tags_dict[key]
+
+    return tags_to_set, tag_keys_to_unset
+
+
+def tag_list_to_dict(tag_list):
```

Comment:
Same for `boto3_tag_list_to_ansible_dict`:
https://github.com/ansible-collections/amazon.aws/blob/main/plugins/module_utils/ec2.py#L465